core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -457,18 +457,25 @@ class KafkaZkClient private[zk] (zooKeeperClient: ZooKeeperClient, isSecure: Boo
 
   /**
    * Gets all topics in the cluster.
+   * @param registerWatch indicates if a what must be registered or not
    * @return sequence of topics in the cluster.
    */
-  def getAllTopicsInCluster: Set[String] = {
-    val getChildrenResponse = retryRequestUntilConnected(GetChildrenRequest(TopicsZNode.path))
+  def getAllTopicsInCluster(registerWatch: Boolean): Set[String] = {
+    val getChildrenResponse = retryRequestUntilConnected(
+      GetChildrenRequest(TopicsZNode.path, registerWatch))
     getChildrenResponse.resultCode match {
       case Code.OK => getChildrenResponse.children.toSet
       case Code.NONODE => Set.empty
       case _ => throw getChildrenResponse.resultException.get
     }
-
   }
 
+  /**
+   * Gets all topics in the cluster. It does not register a watch.
+   * @return sequence of topics in the cluster.
+   */
+  def getAllTopicsInCluster(): Set[String] = getAllTopicsInCluster(false)
```

Comment:
Overloads like this are generally not done in Scala since you can use default arguments. But maybe it's best to explicit here.

Reply:
Right. I have kept this method one because it is called from Java as well where default arguments do not work.

Reply:
What Java code?

Reply:
It is used in `org.apache.kafka.streams.integration.utils.EmbeddedKafkaCluster`.

Reply:
We could use the other method there and pass `false` but I felt that it is more convenient like this.

Reply:
I have removed it finally. It is cleaner.